resources/tentative/todomvc/src/components/todo-list/todo-list.component.js

```diff
@@ -0,0 +1,123 @@
+import template from "./todo-list.template.js";
+import TodoItem from "../todo-item/todo-item.component.js";
+
+class TodoList extends HTMLElement {
+    static get observedAttributes() {
+        return ["total-items"];
+    }
+
+    constructor() {
+        super();
+        // state
+        this._elements = [];
+        this._route = undefined;
+        // elements
+        const node = document.importNode(template.content, true);
+        this.listNode = node.querySelector(".todo-list");
+        // create shadow dom
+        this.shadow = this.attachShadow({ mode: "open" });
+        // rtl support to target with styles
+        this.htmlDirection = document.querySelector("html").getAttribute("dir") || "ltr";
+        this.shadow.host.setAttribute("dir", this.htmlDirection);
+        // add shadow dom
+        this.shadow.append(node);
+    }
+
+    addItem(entry) {
+        const element = new TodoItem();
+        Object.keys(entry).forEach(key => element.setAttribute(key, entry[key]));
+
+        this._elements.push(element);
+        this.listNode.append(element);
+    }
+
+    addItems(items) {
+        items.forEach((entry) => this.addItem(entry));
+    }
+
+    removeCompletedItems() {
+        this._elements = this._elements.filter((element) => {
+            if (element.completed === "true")
+                element.removeItem();
+
+            return element.completed === "false";
+        });
+    }
+
+    toggleItems(completed) {
+        this._elements.forEach((element) => {
+            if (completed && element.completed === "false")
+                element.toggleInput.click();
+            else if (!completed && element.completed === "true")
+                element.toggleInput.click();
+
+        });
+    }
+
+    updateStyles() {
+        if (parseInt(this["total-items"]) !== 0)
+            this.listNode.style.display = "block";
+        else
+            this.listNode.style.display = "none";
+
+    }
+
+    updateView(element) {
+        switch (this._route) {
+            case "completed":
+                element.style.display
+                    = element.completed === "true" ? "block" : "none";
```

Comment:
It seems like this can fit in a single line??

Reply:
fixed